cmd/infra/powervs/destroy.go

```diff
@@ -371,8 +376,64 @@ func deleteSecrets(name, namespace, cloudInstanceID string, accountID string, re
 	return nil
 }
 
+// destroyPowerVsDhcpServer destroying powervs dhcp server
+func destroyPowerVsDhcpServer(ctx context.Context, logger logr.Logger, cloudInstanceID string, session *ibmpisession.IBMPISession) error {
+	client := instance.NewIBMPIDhcpClient(ctx, session, cloudInstanceID)
+
+	dhcpServers, err := client.GetAll()
+	if err != nil {
+		return err
+	}
+
+	if len(dhcpServers) == 0 {
+		logger.Info("No DHCP servers available to delete in PowerVS")
+		return nil
+	}
+
+	dhcpID := *dhcpServers[0].ID
+	logger.Info("Deleting DHCP server", "id", dhcpID)
+	err = client.Delete(dhcpID)
+	if err != nil {
+		return err
+	}
+
+	instanceClient := instance.NewIBMPIInstanceClient(ctx, session, cloudInstanceID)
+
+	// TO-DO: need to replace the logic of waiting for dhcp service deletion by using jobReference.
+	// jobReference is not yet added in SDK
+	f := func() (bool, error) {
+		dhcpInstance, err := instanceClient.Get(dhcpID)
+		if err != nil {
+			if err = isNotRetryableError(err, timeoutErrorKeywords); err == nil {
+				return false, nil
+			}
+			errMsg := err.Error()
+			// when instance becomes does not exist, infra destroy can proceed
+			// when there is flaky, internal server error in pvm-instance get call, retry destroy
+			if strings.Contains(errMsg, "pvm-instance does not exist") || strings.Contains(errMsg, "unable to get the server volume attachment list: Internal Server Error") {
+				err = nil
+				return true, nil
+			}
+			return false, err
+		}
+
+		if dhcpInstance == nil {
+			return false, fmt.Errorf("dhcpInstance is nil")
+		}
+
+		logger.Info("Waiting for DhcpServer to destroy", "id", *dhcpInstance.PvmInstanceID, "status", *dhcpInstance.Status)
+		if *dhcpInstance.Status == dhcpInstanceDeletedState || *dhcpInstance.Status == dhcpServiceErrorState {
```

Comment:
Why are you considering `dhcpServiceErrorState` here?

Reply:
Removed it!
I actually took reference from your PR - https://github.com/openshift/hypershift/pull/2867/files#diff-f4fe34c4d57d805e9983ff19def305eeb4bd6ab244f96b4cc66076361ed478d4L388
Thought this check is added, so that the PowerVS instance deletion can proceed even if the dhcp server deletion is in error state, as anyways, the DHCP server will get deleted as  PowerVS instance recursive delete operation.